packages/shared/src/contexts/ActiveFeedContext.tsx

```diff
@@ -1,11 +1,118 @@
-import { createContext } from 'react';
+import React, {
+  ReactElement,
+  ReactNode,
+  useContext,
+  useMemo,
+  useRef,
+} from 'react';
 import { FeedReturnType } from '../hooks/useFeed';
+import { Post } from '../graphql/posts';
+import useLeanPostActions from '../hooks/post/useLeanPostActions';
 
 export type ActiveFeedContextValue = {
   queryKey?: unknown[];
+  pinPost?: boolean;
   items: FeedReturnType['items'];
+  onClick?: (post: Post) => void;
+  onBookmark?: (post: Post) => void;
+  onUpvote?: (post: Post) => void;
+  onDownvote?: (post: Post) => void;
+  onHidePost?: (post: Post) => void;
+  onPromotePost?: (post: Post) => void;
+  onBanPost?: (post: Post) => void;
+  onDeletePost?: (post: Post) => void;
+  canDeletePost?: (post: Post) => boolean;
+  canPinPost?: (post: Post) => boolean;
+  onPinPost?: (post: Post) => void;
+  onBlockSource?: (post: Post) => void;
+  onBlockTag?: (post: Post, tag: string) => void;
+  feedRef?: React.RefObject<HTMLDivElement>;
 };
 
-export const ActiveFeedContext = createContext<ActiveFeedContextValue>({
+const ActiveFeedContext = React.createContext<ActiveFeedContextValue>({
   items: [],
 });
+
+export default ActiveFeedContext;
+
+type ActiveFeedContextProviderProps = ActiveFeedContextValue & {
+  children: ReactNode;
+};
+
+export const ActiveFeedContextProvider = ({
+  children,
+  items,
+  queryKey,
+  onClick,
+  pinPost = true,
+}: ActiveFeedContextProviderProps): ReactElement => {
+  const feedRef = useRef();
+  const virtualGrid =
+    feedRef.current && getComputedStyle(feedRef.current).gridTemplateColumns;
+
+  // TODO: determine tracking when lean post actions are invoked from a feed type
+  const {
+    onPromotePost,
+    onBanPost,
+    canDeletePost,
+    onDeletePost,
+    canPinPost,
+    onPinPost,
+    onBlockSource,
+    onBlockTag,
+    onClick: onClickDefault,
+    onBookmark,
+    onUpvote,
+    onDownvote,
+    onHidePost,
+  } = useLeanPostActions({
+    queryKey,
+  });
+
```

Comment:
Why do we need this top level, it should be called just in nested components?

Reply:
This can actually move into the nested components indeed.
The only exposure here I think was in case you can't to overwrite handlers?

Not sure if it's needed for that.